js/actions/sources.js

```diff
@@ -66,6 +66,14 @@ function loadSources() {
     [PROMISE]: Task.spawn(function*() {
       const response = yield gThreadClient.getSources();
 
+      gThreadClient.addListener("newSource", (event, packet) => {
+        store.dispatch(actions.newSource(packet.source));
```

Comment:
`loadSources` is an action that means "get all the sources for the page". It definitely should not install listeners. There's no reason I can't call `loadSources` multiple times, and the current debugger does just that when switching sourcemaps on/off. I'd rather keep everything in `main.js` for now and next week we can clean up the protocol part.

I don't think it's too much coupling. For things outside the react/redux system, it's a simple pattern for them to "hook up" to it by directly dispatching actions on the store. It makes it very clear what "drives" the system. An initial action (not one dispatched from within an action creator) can *only* be dispatched from two places: a UI event from a component, or a network event from some external source. I think it's nice to be able to see how the external source "binds" itself to the system: in this one file, all of the network events map to these actions.

The core issue is that actions can be called multiple times, so you don't want to have side effects in there that shouldn't happen multiple times.